examples/fetch_ic/test/tests.ts

```diff
@@ -0,0 +1,318 @@
+// TODO allow running these tests with and without the authorization header
+// TODO these seem like really good tests for the server and normal fetch too
+// TODO consider how we'll do the axios tests as well
+
+import * as dns from 'node:dns';
+dns.setDefaultResultOrder('ipv4first');
+
+import { Principal } from '@dfinity/principal'; // TODO get rid of this dependency if we installed it
+import { Test, getCanisterOrigin } from 'azle/test';
+import puppeteer, { Browser, Page } from 'puppeteer';
+
+export function getTests(canisterName: string): Test[] {
```

Comment:
I'd like to see a test for headers represented by a Header object

Reply:
Headers is not supported

Reply:
We might need to throw on this, check later PRs...I thought I addressed this or thought about it

Reply:
We do throw an error on Headers: fetchIc: Headers is not a supported headers type